x-pack/plugins/canvas/public/components/workpad_page/index.js

```diff
@@ -4,53 +4,54 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import isEqual from 'react-fast-compare';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import { compose, withState, withProps, withHandlers } from 'recompose';
-import { aeroelastic } from '../../lib/aeroelastic_kibana';
-import { removeElements, insertNodes, elementLayer } from '../../state/actions/elements';
-import { getFullscreen, canUserWrite } from '../../state/selectors/app';
-import { getNodes, isWriteable } from '../../state/selectors/workpad';
-import { flatten } from '../../lib/aeroelastic/functional';
+import { branch, compose, withHandlers, withProps, withState, shouldUpdate } from 'recompose';
+import { elementLayer, insertNodes, removeElements } from '../../state/actions/elements';
+import { canUserWrite, getFullscreen } from '../../state/selectors/app';
+import { getNodes, getPageById, isWriteable } from '../../state/selectors/workpad';
+import { updater } from '../../lib/aeroelastic/layout';
```

Comment:
nit: I know you pointed this out in review, but I wanted to point out that it breaks from convention. `index` exports are considered the "public" API, and now in Aero, index exports nothing and you need to reach inside the library. It probably doesn't matter a ton, but I thought it was worth mentioning.

Reply:
Thanks for making this convention explicit to me. In the current case, what started as a separate repo (`elastic/aeroelastic`) became more of a toolbox of independent things. Eg. the `gesture`/`layout` things are independent of the `store`, one can use one without the other, so I've plans to flatten it out eventually (as happened with the `dom` util for converting a `transformMatrix` into a CSS transform matrix). Same with `matrix.ts`. I'm planning to phase out the `store` itself as soon as we can.

A more technical issue is that on one hand, it's nice for TS type specifications to live in their own file (as now, `index.d.ts`) but weird things arise if there's both an `index.js` (or `index.ts`) and an `index.d.ts`. 

What's your advice, should I just forget about trying to have a dedicated file for the types?

Reply:
What's here is fine, just pointing out the difference. Sounds like your longer vision might be to move all this stuff into `lib`, and drop the Aeroelastic part. And if that's the case, this seems like a reasonable short-term change.

Reply:
Yes, that's the plan!